chia/wallet/wallet_merkle_coin_store.py

```diff
@@ -0,0 +1,265 @@
+from __future__ import annotations
+
+import sqlite3
+from typing import Dict, List, Optional, Set
+
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.db_wrapper import DBWrapper2, execute_fetchone
+from chia.util.ints import uint32, uint64
+from chia.wallet.util.wallet_types import WalletType
+from chia.wallet.wallet_merkle_coin_record import WalletMerkleCoinRecord
+
+
+class WalletMerkleCoinStore:
```

Comment:
a lot of the methods in this class are not used anywhere, why were they added?

Reply:
is there a way to reuse `WalletCoinStore` to some extent so we at least get the wallet coin store test coverage too

Reply:
The only way to reuse the WalletCoinStore is adding new columns to the coin_records table. Create a side table just for merkle coins need to rewrite all CRUD methods.

Reply:
It also has a metadata column that stores coin type specific data. For the performance and recording reasons, merkle coins will not have a copy in the normal coins table.